py/genv/remote/ssh.py

```diff
@@ -102,24 +63,21 @@ async def run(
 
     :return: Returns the hosts that succeeded and their standard outputs
     """
-    processes = [
-        await start(
-            host,
-            command,
-            asyncio.subprocess.PIPE if stdins else asyncio.subprocess.DEVNULL,
-        )
-        for host in config.hosts
-    ]
+    ssh_runners_and_inputs = []
+    for host, stdin in zip(config.hosts, stdins or [None for _ in config.hosts]):
+        ssh_runner = SshRunner(host.hostname, host.timeout, {'PATH': SshRunner.calc_remote_path_env(host.root)})
+        ssh_runners_and_inputs.append((ssh_runner, stdin))
```

Comment:
this is totally sababa but a bit less than the pattern we have been using so far of having separate lists and using `zip()`
so my suggestions is to rewrite this a bit like this:
```
ssh_runners  = [SshRunner(host.hostname, host.timeout, {'PATH': SshRunner.calc_remote_path_env(host.root)}) for host in config.hosts]

ssh_outputs = await asyncio.gather(*(ssh_runner.run(..., stdin) for ssh_runner, stdin in zip(ssh_runners, stdins or [None for _ in config.hosts])))
```

let me know what you think
it's just that now we are using this `zip` pattern all over the place and not plain `for`-loops